### PR TITLE
Typescript, isCompleted

### PR DESCRIPTION
**Pull request by CrispusDH**
- add typescript
- implement isCompleted in own way (for further using in crossing ToDo item)

### CODE REVIEW

File: src/Root.tsx [hunk 1]
```diff
@@ -1,12 +1,12 @@
-import React from 'react';
 import { List } from './component/list'
-import { Headline } from "./component/headline";
+import { Headline } from './component/headline';
+import * as React from 'react'
```
Comment (sashkhen) [inline src/Root.tsx]: does this change have purpose?)

File: src/component/form.tsx [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+export function Form(props: Props) {
+  const initial = '';
+  const [value, setValue] = useState(initial);
+
+  const onChange = (event: any) => {
+    setValue(event.target.value)
+  };
+
+  const onSubmit = (event: any) => {
```
Comment (sashkhen) [inline src/component/form.tsx]: 👍

File: src/component/form.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+  public isComplete: boolean;
+
+  constructor(text: string) {
+    this.text = text;
+    this.id = uuid();
+    this.isComplete = false;
+  }
+
+  toggleComplete() {
+    this.isComplete = !this.isComplete;
```
Comment (sashkhen) [inline src/component/form.tsx]: 👍

File: src/component/list.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+  const left = () => {
+    return todos.filter((todo) => !todo.isComplete).length;
+  };
+
+  const filtered = () => {
+    let array: ToDo[] = [];
+    switch (filter) {
+      case FilterOptions.All: {
+        array = todos;
```
Comment (sashkhen) [inline src/component/list.tsx]: you can use ‘return todos.filter(...’ without assigning it to a variable and breaks and its also immutable

File: src/component/list.tsx [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+        <button onClick={() => setFilter(FilterOptions.All)}>all</button>
+        <button onClick={() => setFilter(FilterOptions.Active)}>active</button>
+        <button onClick={() => setFilter(FilterOptions.Completed)}>completed</button>
+      </div>
+    </div>
+  )
+}
+
+export enum FilterOptions {
+  All,
```
Comment (sashkhen) [inline src/component/list.tsx]: how does it work? cant see any of the keys being defined

**Comment by sashkhen** [comment]
meow

File: src/component/list.tsx (hunk 5 above)
Comment (CrispusDH) [reply src/component/list.tsx]: take a look at [it link](https://basarat.gitbooks.io/typescript/docs/enums.html)

File: src/Root.tsx (hunk 1 above)
Comment (CrispusDH) [reply src/Root.tsx]: no, I see error during indexing. Currently, I will return it back

File: src/component/form.tsx (hunk 2 above)
Comment (CrispusDH) [reply src/component/form.tsx]: I will check what type should be on event

File: src/component/list.tsx (hunk 4 above)
Comment (CrispusDH) [reply src/component/list.tsx]: Good catch, thanks!

**Pull request merged**